src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
 import java.awt.Color;
 import java.awt.Dimension;
 import java.awt.Font;
-import java.awt.Graphics;
-import java.awt.Graphics2D;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import java.awt.event.ComponentAdapter;
 import java.awt.event.ComponentEvent;
 import java.awt.event.KeyEvent;
 import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
+import java.io.IOException;
+import java.net.URL;
 
 import javax.swing.BoxLayout;
 import javax.swing.JFrame;
 import javax.swing.JPanel;
 import javax.swing.JScrollPane;
-import javax.swing.JTextArea;
 import javax.swing.JTextField;
+import javax.swing.JTextPane;
 import javax.swing.KeyStroke;
-import javax.swing.plaf.basic.BasicTextAreaUI;
 import javax.swing.text.BadLocationException;
-import javax.swing.text.Document;
-import javax.swing.text.Element;
-import javax.swing.text.JTextComponent;
-import javax.swing.text.PlainView;
-import javax.swing.text.Segment;
-import javax.swing.text.TabExpander;
-import javax.swing.text.Utilities;
-import javax.swing.text.View;
-import javax.swing.text.WrappedPlainView;
+import javax.swing.text.html.HTMLDocument;
+import javax.swing.text.html.*;
 
 import VASSAL.build.Buildable;
 import VASSAL.build.GameModule;
 import VASSAL.command.Command;
 import VASSAL.command.CommandEncoder;
 import VASSAL.configure.ColorConfigurer;
 import VASSAL.configure.FontConfigurer;
+import VASSAL.configure.BooleanConfigurer;
 import VASSAL.i18n.Resources;
 import VASSAL.preferences.Prefs;
 import VASSAL.tools.ErrorDialog;
 import VASSAL.tools.KeyStrokeSource;
 import VASSAL.tools.ScrollPane;
-import VASSAL.tools.swing.SwingUtils;
+
+//BR// 23-June-2020 - significant Chatter update to provide full HTML support, including multiple colors/fonts and images, <b>bold</b> and <i>italic</i> 
```

Comment:
This is handled by the source code management / by git in this case, it keeps track of who changed what and when, and if the commit message contains a bug ID then it's also possible to look it up later on and find out why something was changed.

Reply:
fixed for next push

Reply:
fixed for next push